src/artemis/ispyb/store_in_ispyb.py

```diff
@@ -47,7 +47,7 @@ def begin_deposition(self):
         ) = self.store_grid_scan(self.full_params)
         return self.datacollection_ids, self.grid_ids, self.datacollection_group_id
 
-    def end_deposition(self, success):
+    def end_deposition(self, success, reason):
```

Comment:
Nit: This might be a bit of overkill but I think it would be nice to create some kind of `ISpyBStatus` enum here, a bit like we do in [main](https://github.com/DiamondLightSource/python-artemis/blob/59739c26fdd3a99c853a8b96f3804cd945c5c9b1/src/artemis/__main__.py#L43) along with a helper function of `bluesky_doc_to_ispyb_status`. This has a couple of advantages:
* Pulls the magic strings of `DataCollection Successful` into an obvious constant
* Separates the interface for this class and for bluesky so we're not conflating them. e.g. the use of `fail` in here is a bluesky thing but not obviously

Reply:
What are your thoughts?